Snowflake.Data/Core/Session/SFSession.cs

```diff
@@ -375,6 +388,30 @@ internal void UpdateSessionParameterMap(List<NameValueParameter> parameterList)
                     stopHeartBeatForThisSession();
                 }
             }
+            if (ParameterMap.ContainsKey(SFSessionParameter.QUERY_CONTEXT_CACHE_SIZE))
+            {
+                string val = ParameterMap[SFSessionParameter.QUERY_CONTEXT_CACHE_SIZE].ToString();
+                queryContextCacheSize = Int32.Parse(val);
+                queryContextCache.setCapacity(queryContextCacheSize);
+            }
+        }
+
+        internal void UpdateQueryContextCache(ResponseQueryContext queryContext)
+        {
+            if (disableQueryContextCache)
+            {
+                return;

```

Comment:
we could debug that it is disabled in case anyone wonders why it's not getting context, does it make sense?

Reply:
done